keps/sig-node/2411-cri-container-log-rotation/README.md

```diff
@@ -0,0 +1,301 @@
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a kubernetes user, I want to use a CRI container runtime and want the container logs to be managed and rotated by the kubelet.
+
+#### Story 2
+
+As a kubernetes user, I want to integrate a logging agent that aggregates the logs to a remote storage without having to further integrate with the CRI container runtime.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+- Loss of some logs during log rotation. There is an open issue on this with a suggested fix https://github.com/kubernetes/kubernetes/issues/64760
+
+## Design Details
+
+This implementation adds container log manager package, which the kubelet uses to manage and rotate the logs. The container log manager will only start up when the container runtime being used is one that communicates with the kubelet via the CRI i.e CRI-O, containerd, etc.
+
+The rotated logs are compressed with gzip. The latest rotated log is not compressed as a logging agent, such as fluentd, might still be reading it right after rotation. The kubelet periodically checks the amount of disk space being used by the container logs and rotates them if the max value has been reached. After the logs are rotated, the kubelet sends a signal to the container runtime to re-open the log file.
+
+The user can configure the maximum size of a log file and the maximum number of log files to retain with the **--container-log-max-size** and **--container-log-max-files** flags. The default values are **10Mi** for the max file size and **5** for the max number of log files.
+
+The kubelet exposes a consistent log directory structure with embedded metadata so that logging agents can integrate with it is. The logging agents can directly work with the kubelet without having to make any further integrations with the container runtime in use.
+
+### Test Plan
+
+- There are currently unit tests and node E2E integration tests for container log rotation
+- Get feedback on performance and stability of the CRI log format on other products aside from OpenShift and GKE.
+
+### Graduation Criteria
+
+#### Alpha -> Beta Graduation
+
+- Unit and node E2E tests are consistently passing
+- Logging agents can easily integrate with Kubernetes and push rotated logs to a remote storage
+- Successful log rotations by the kubelet
+
+#### Beta -> GA Graduation
+
+- Successfully run in production
+- CRI log format works with other products besides OpenShift and GKE
```

Comment:
I think we should rephrase this:

- Solicit feedback in SIG Node community that there are no issues with individual distributions production usage (OpenShift and GKE both report no major issue)

Given the number of distributions using CRI now in prod, I would have expected more noise if there was a problem...

Reply:
pro-actively soliciting input here:

@jaypipes any known issue from EKS that would raise concern promoting this to GA beyond what is enumerated?
@relyt0925 any known issue from IKS that would raise concern promoting this to GA beyond what is enumerated?

Reply:
Fixed

Reply:
@derekwaynecarr no problems from an IKS perspective!